metals/src/main/scala/scala/meta/internal/metals/codeactions/BracelessBracefulSwitchCodeAction.scala

```diff
@@ -0,0 +1,558 @@
+        path,
+        expectedBraceStartPos = bracePose,
+        expectedBraceEndPose = endBracePose,
+        bracelessStart = "",
+        bracelessEnd = "",
+        indentation = indentation,
+        document = document,
+        codeActionSubjectTitle = codeActionSubjectTitle
+      )
+    }
+  }
+
+  /**
+   * @param blockHolder      the trees which can only be braced if the type of a branch of
+   *                         them is `Term.Block` and that block is braced.
+   * @param path             the path to the file containing the tree. It is used in
+   *                         contstructing the TextEdit of the code action
+   * @param document
+   * @param blockEmbraceable the branch of the main tree which can potentially have
+   *                         the type `Term.Block`; so can be braced.
+   * @param codeActionSubjectTitle
+   * @return
+   */
+  private def createCodeActionForPotentialBlockHolder(
+      blockHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      blockEmbraceable: Term,
+      codeActionSubjectTitle: String
+  ): Option[l.CodeAction] = { // the block embraceable
+    val indentation =
+      getIndentationForPositionInDocument(blockHolder.pos, document)
+
+    if (isBlockEmbraceableBraced(blockEmbraceable)) {
+      for {
+        bracePose <- util
+          .Try(blockEmbraceable.tokens.minBy(_.pos.start))
+          .toOption
+          .map(_.pos.toLSP.getStart)
+        if blockEmbraceable.allowBracelessSyntax
+      } yield createCodeActionForGoingBraceless(
+        path,
+        expectedBraceStartPos = bracePose,
+        expectedBraceEndPose = blockEmbraceable.pos.toLSP.getEnd,
+        bracelessStart = "",
+        bracelessEnd = "",
+        codeActionSubjectTitle
+      )
+
+    } else { // does not have braces
+      for {
+        bracePose <- util
+          .Try(blockEmbraceable.tokens.minBy(_.pos.start))
+          .toOption
+          .map(_.pos.toLSP.getStart)
+      } yield createCodeActionForGoingBraceful(
+        path,
+        expectedBraceStartPos = bracePose,
+        expectedBraceEndPose = blockHolder.pos.toLSP.getEnd,
+        bracelessStart = "",
+        bracelessEnd = "",
+        indentation = indentation,
+        document = document,
+        codeActionSubjectTitle = codeActionSubjectTitle
+      )
+    }
+  }
+
+  private def createCodeActionForTemplateHolder(
+      templateHolder: Tree,
+      path: AbsolutePath,
+      document: String,
+      templ: Template,
+      codeActionSubjectTitle: String
+  ): Option[l.CodeAction] = {
+    val indentation =
+      getIndentationForPositionInDocument(templateHolder.pos, document)
+
+    util.Try(templ.inits.maxBy(_.pos.end).pos.end).toOption
+    val expectedBraceStartPos = util
+      .Try {
+        val lastInit = templ.inits.maxBy(init => init.pos.end)
+        lastInit.pos.end
+      }
+      .getOrElse(templ.pos.start)
+    pprint.log("templ.inits: \n" + templ.inits)
```

Comment:
pprint is not available on CI, so this might break the compilation

Reply:
Sure. Just for debugging.

Reply:
I wanted to let you know, since the compilation issue is not obvious :sweat_smile:

Reply:
Ok, thank you so much!